pkg/sql/catalog/lease/descriptor_version_state.go

```diff
@@ -131,9 +139,26 @@ func (s *descriptorVersionState) incRefCountLocked(ctx context.Context, expensiv
 	}
 }
 
+func (s *descriptorVersionState) getExpirationAndHasSession() (hlc.Timestamp, bool) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	// If a session is stored in the version state, then the expiry
+	// will be derived from the session.
+	if s.mu.session != nil {
+		return s.mu.session.Expiration(), true
+	}
+	return s.mu.expiration, false
+}
+
 func (s *descriptorVersionState) getExpiration() hlc.Timestamp {
```

Comment:
I suggest get rid of this method and just use the new `getExpirationAndHasSession` instead, you may always ignore the second return value.